### DISCUSSION TITLE
Extend /api/upload to allow folders

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [diced/zipline#528] [Ideas]
Hello!

I am working on a project for my company to store NixOS configurations sent from remote monitoring agents - for this, I would like to be able to organize the various files uploaded into folders.

Consider the following aspects:
- Username = `$HOSTNAME`
- Password = Generated beforehand, becomes part of the per-device configuration
- Files:
  * `/etc/nixos/configuration.nix` -> `$(date "...")/configuration.nix`
  * `/etc/sshd/ssh_host_*` -> `ssh_host_*` (deny if exists)
  * And possibly others, some sorted into a `$(date "...")` subfolder.

The aim is to archive successful configratuions for those devices using NixOS' `system.activationScripts` property by running cURL to upload the files.

I see that folder support exists, but I have no idea how to upload to those folders...

Thanks and kind regards!

/cc @senpro-ingwersenk (thats me, but my work account, so I can track it entirely.)

**Comment by diced** [comment]
ima just add this in because v4 will take a bit longer to release 😅

**Comment by diced** [comment]
Create a folder on the dashboard, then use its ID within the `x-zipline-folder` header. If a folder has an ID of 4, then the header would be `x-zipline-folder: 4`. Support for this was added in https://github.com/diced/zipline/commit/1a837c02d2352105e9632d6a052d7cb6aa2b9fc1
